src/tiffimage_int.cpp

```diff
@@ -1046,13 +1190,24 @@ namespace Exiv2 {
         { Tag::root, fujiId,           exifId,           0x927c    },
         { Tag::root, canonId,          exifId,           0x927c    },
         { Tag::root, canonCsId,        canonId,          0x0001    },
-        { Tag::root, canonSiId,        canonId,          0x0004    },
+        { Tag::root, canonSiId,        canonId,          0x0004    },        
+        { Tag::root, canonAf2Id,       canonId,          0x0026    },
```

Comment:
I don't think that this works, because the AF field of canon maker notes is handled in a special way.   
See [line 1816](https://github.com/Exiv2/exiv2/blob/a61853b32308c1a0eadc38ae838aa931bf2c388b/src/tiffimage_int.cpp#L1816) where we define a special `TiffDecoder` for this element.   

That's why I think that we can actually remove most (maybe all) the AF stuff from `canonmn_int.{hpp,cpp}`

Reply:
Actually I see the following: 

Exif.CanonAf2Id.AFInfoSize                   SLong       1  1216
Exif.CanonAf2Id.AFAreaMode                   SLong       1  (65679)
Exif.CanonAf2Id.AFNumPoints                  SLong       1  293608000
Exif.CanonAf2Id.AFValidPoints                SLong       1  293608000
Exif.CanonAf2Id.AFCanonImageWidth            SLong       1  468
Exif.CanonAf2Id.AFCanonImageHeight           SLong       1  0
Exif.CanonAf2Id.AFImageWidth                 SLong       1  0
Exif.CanonAf2Id.AFImageHeight                SLong       1  0
Exif.CanonAf2Id.AFAreaWidths                 SLong       1  0
Exif.CanonAf2Id.AFAreaHeights                SLong       1  0
Exif.CanonAf2Id.AFXPositions                 SLong       1  0
Exif.CanonAf2Id.AFYPositions                 SLong       1  0
Exif.CanonAf2Id.AFPointsInFocus              SLong       1  0
Exif.CanonAf2Id.AFPointsSelected             SLong       1  0
Exif.CanonAf2Id.AFPointsUnusable             SLong       1  0
Exif.CanonAf2Id.AFFineRotation               SLong       1  0

And: 

Exif.CanonAfC.AFConfigTool                   SLong       1  0
Exif.CanonAfC.AFTrackingSensitivity          SLong       1  2
Exif.CanonAfC.AFAccelDecelTracking           SLong       1  2
Exif.CanonAfC.AFPointSwitching               SLong       1  0
Exif.CanonAfC.AIServoFirstImage              SLong       1  Equal Priority
Exif.CanonAfC.AIServoSecondImage             SLong       1  Equal Priority
Exif.CanonAfC.USMLensElectronicMF            SLong       1  Disable in AF Mode
Exif.CanonAfC.AFAssistBeam                   SLong       1  Disable
Exif.CanonAfC.OneShotAFRelease               SLong       1  Release Priortiy
Exif.CanonAfC.AutoAFPointSelEOSiTRAF         SLong       1  Enable
Exif.CanonAfC.LensDriveWhenAFImpossible      SLong       1  Continue Focus Search
Exif.CanonAfC.SelectAFAreaSelectionMode      SLong       1  (22359)
Exif.CanonAfC.AFAreaSelectionMethod          SLong       1  M-Fn Button
Exif.CanonAfC.OrientationLinkedAF            SLong       1  Same for Vert/Horiz Points
Exif.CanonAfC.ManualAFPointSelPattern        SLong       1  Stops at AF Area Edges
Exif.CanonAfC.AFPointDisplayDuringFocus      SLong       1  Selected (constant)
Exif.CanonAfC.VFDisplayIllumination          SLong       1  Auto
Exif.CanonAfC.AFStatusViewfinder             SLong       1  Auto
Exif.CanonAfC.InitialAFPointInServo          SLong       1  Initial AF Point Selected

Reply:
Exif.CanonAfC.SelectAFAreaSelectionMode SLong 1 (22359) --> This tag needs some more work though

Reply:
Exif.CanonAf2Id.AFAreaMode SLong 1 (65679) --> This could also need some digging